sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClientBuilder.java

```diff
@@ -0,0 +1,311 @@
+            ? Configuration.getGlobalConfiguration()
+            : configuration;
+
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> httpPipelinePolicies = new ArrayList<>();
+        httpPipelinePolicies.add(new AddHeadersPolicy(headers));
+        httpPipelinePolicies.add(new RequestIdPolicy());
+
+        HttpPolicyProviders.addBeforeRetryPolicies(httpPipelinePolicies);
+        httpPipelinePolicies.add(retryPolicy == null ? new RetryPolicy() : retryPolicy);
+
+        httpPipelinePolicies.add(new AddDatePolicy());
+        if (keyCredential != null) {
+            this.policies.add(new AzureKeyCredentialPolicy(API_KEY, keyCredential));
+        }
+        httpPipelinePolicies.addAll(this.policies);
+
+        HttpPolicyProviders.addAfterRetryPolicies(httpPipelinePolicies);
+
+        httpPipelinePolicies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
+            buildConfiguration));
+        httpPipelinePolicies.add(new HttpLoggingPolicy(httpLogOptions));
+
+        HttpPipeline buildPipeline = new HttpPipelineBuilder()
+            .httpClient(httpClient)
+            .policies(httpPipelinePolicies.toArray(new HttpPipelinePolicy[0]))
+            .build();
+
+        return new SearchAsyncClient(endpoint, indexName, buildVersion, buildPipeline);
+    }
+
+    /**
+     * Sets the service endpoint for the Azure Search instance.
+     *
+     * @param endpoint The URL of the Azure Search instance.
+     * @return The updated SearchClientBuilder object.
+     * @throws IllegalArgumentException If {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public SearchClientBuilder endpoint(String endpoint) {
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL"));
+        }
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the {@link AzureKeyCredential} used to authenticate HTTP requests.
+     *
+     * @param keyCredential The {@link AzureKeyCredential} used to authenticate HTTP requests.
+     * @return The updated SearchClientBuilder object.
+     * @throws NullPointerException If {@code keyCredential} is {@code null}.
+     * @throws IllegalArgumentException If {@link AzureKeyCredential#getKey()} is {@code null} or empty.
+     */
+    public SearchClientBuilder credential(AzureKeyCredential keyCredential) {
+        if (keyCredential == null) {
+            throw logger.logExceptionAsError(new NullPointerException("'keyCredential' cannot be null."));
+        }
+        if (CoreUtils.isNullOrEmpty(keyCredential.getKey())) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("'keyCredential' cannot have a null or empty API key."));
```

Comment:
Should update the exception message here as `null` with throw a different exception.